### PR TITLE
docs: change docs for get request return object type checking

### LINKED ISSUE #34746: Tour of Heroes (toh-pt6): Misleading information on HTTP request response type
# 📚 Docs or angular.io bug report

### Description

In section "HttpClient.get() returns response data" the tutorial says:
"HttpClient.get() returns the body of the response as an untyped JSON object by default. Applying the optional type specifier, <Hero[]> , gives you a typed result object."

For me this sentence was very misleading when learning Angular. I was expecting to be able to provide a class type and would get an object instance including all the members of the class - specifically I would be able to call mehods defined in the class.

The class type is only used for syntax check / code-completion, however, during runtime the object returned by the get() method is a plain javascript object as received with the JSON response from the server, with the prototype being "Object". It does not contain any further members of the class.

The tutorial should explain this in more detail, and if there is any best practices on how to transform from the JSON object to a real typed object, this should also be explained or pointed to.

The functionality/feature that a plain JSON object is returned despite of the type being specified is for example discussed here:
https://github.com/angular/angular/issues/20770

### What's the affected URL?**
https://angular.io/tutorial/toh-pt6#httpclientget-returns-response-data

### PR DESCRIPTION
**Pull request by ajitsinghkaler**
earlier in the docs it was told that adding typescript's type checking makes JSON response typed so changed docs to say that it adds all typescript capabilities during compile time

Fixes #34746

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

earlier in the docs it was told that adding typescript's type checking makes JSON response typed

Issue Number: #34746 

## What is the new behavior?
changed docs to say that it adds all typescript capabilities during compile time

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ajitsinghkaler** [comment]
@kapunahelewong can you please have a look

File: aio/content/tutorial/toh-pt6.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ This particular `HttpClient.get()` call returns an `Observable<Hero[]>`; that is
 ### `HttpClient.get()` returns response data
 
 `HttpClient.get()` returns the body of the response as an untyped JSON object by default.
-Applying the optional type specifier, `<Hero[]>` , gives you a typed result object.
+Applying the optional type specifier, `<Hero[]>` , adds all the typescript capabilities which helps in reducing a lot of errors during compile time.
```
Comment (kapunahelewong) [inline aio/content/tutorial/toh-pt6.md]: How about this?
```suggestion
Applying the optional type specifier, `<Hero[]>` , adds TypeScript capabilities, which reduce errors during compile time.
```

**Review by kapunahelewong — CHANGES_REQUESTED** [review]
Thanks so much for this, @ajitsinghkaler. Just a small suggestion.

**Comment by ajitsinghkaler** [comment]
@kapunahelewong I've made the changes please have a look

**Review by kapunahelewong — APPROVED** [review]
LGTM!

**Comment by kapunahelewong** [comment]
merge-assistance: Global approval for this docs-only change.

**Pull request closed without merging**